### PR TITLE
Adding a test for fuzzy operator

### PR DESCRIPTION
**Pull request by alexeyche**
Forgive my Ruby, this is my first attempt writing in it.

This PR is a part of the work on propagating annotations for fuzzy:
https://github.com/vespa-engine/vespa/pull/21972

It's dependent on this PR, so this PR should be merged only after that PR

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: tests/search/fuzzy_matching/fuzzy.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      run_fuzzysearch_max_edit_tests(f)
+    }
+    run_fuzzysearch_prefix_length_tests("array_fast")
+    run_fuzzysearch_prefix_length_tests("array_slow")
+  end
+
+  def run_fuzzysearch_default_tests(f)
+    puts "Running fuzzy search tests with default parameters: #{f}"
+
+    assert_hitcount(make_query(make_fuzzy(f, "ThisIsAFox")), 5)
```
Comment (geirst) [inline tests/search/fuzzy_matching/fuzzy.rb:76]: For more robust testing, consider checking that the expected documents are returned. This can be done using a document summary in the schema with the built-in field _documentid_:
```
document-summary only_id {
    summary documentid type string {}
}
```
, and specifying that it should be used in the query (https://docs.vespa.ai/en/reference/query-api-reference.html#presentation.summary). A small helper function can wrap it all together. Something like this:
```
def assert_documents(query, exp_docids)
  result = search(query)
  assert_hitcount(result, exp_docids.size)
  result.sort_results_by("documentid")
  for i in 0...exp_docids.size do
    exp_value = "id:test:test::#{exp_docids[i]}"
    assert_field_value(result, "documentid", exp_docid, i)
  end
end
```

**Review by geirst — COMMENTED** [review]
This looks good. Consider adding some test cases for array and weighted set fields where they contain multiple elements, where only some (or none) match the fuzzy term.

**Comment by alexeyche** [comment]
@geirst Thanks for the review! I am done with addressing comments. Let's also chat about how to disable match for indexing fields, but of course, it should be done in the main repo.

**Review by geirst — APPROVED** [review]
👍

File: tests/search/fuzzy_matching/test.sd [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      cased
+    }
+    match: cased
+  }
+  field single_scan_cased type string {
+    indexing: input single_scan_uncased | attribute
+    match: cased
+  }
+  field single_index type string {
+    indexing: input single_scan_uncased | index
```
[the remark this answers is not in the document]
Comment (geirst) [reply tests/search/fuzzy_matching/test.sd:51]: @alexeyche The validation of FuzzyItem should happen in a validating searcher, similar to container-search/src/main/java/com/yahoo/search/searchers/ValidateNearestNeighborSearcher.java. The searcher should be added to the default search chain, see container-search/src/main/java/com/yahoo/search/searchchain/model/federation/LocalProviderSpec.java.
